### DISCUSSION TITLE
Sample Code to Copy Data to another cube in another model

### DISCUSSION THREAD

**Discussion by chewzauk** [cubewise-code/tm1py#850] [Q&A]
Hi Guys

Does anyone perhaps have sample code to copy data from a cube in one model to a cube in another model?
Kind of like a replication thing...as far as possible anyway.

Many thx
Chris

**Comment by MariusWirtz** [comment]
Hi @chewzauk,

it's quite simple:
- connect to instance A
- Use the `execute_mdx` function or `execute_view` to read data from cube C.
- connect to instance B
- Use the `write` function to write to cube C

Things to consider
- parallelization
- transformations before writing to B
- `use_ti=True` in `write` function to increase performance
- `skip_non_updateable=True` in `write` function 
- synchronize dimensions first before attempting to write

**Comment by chewzauk** [comment]
Thx Marius, much appreciated. Can you provide any advice on how to sync
dimensions...?

**Reply by MariusWirtz** [reply]
Sure:

- connect to instance A
- read the dimension D
- connect to instance B
- write/update the dimension D
- make sure to consider attributes values as well

**Comment by chewzauk** [comment]
Thanks!
